api/event_observer.go

```diff
@@ -211,3 +211,136 @@ func recvEventRequest(
 		return &nb, nil
 	}
 }
+
+// this needs to be greater than the highest eventspb.BusEvent event type
+const maxEventTypeOrdinal = 299
+
+type eventStats struct {
+	eventCount    [maxEventTypeOrdinal + 1]int
+	ignoredEvents bool
+}
+
+func (s *eventStats) updateStats(events []*eventspb.BusEvent) {
+	for _, event := range events {
+		eventType := event.Type
+		if int(eventType) > maxEventTypeOrdinal {
+			eventType = maxEventTypeOrdinal
+		}
+		s.eventCount[eventType] = s.eventCount[eventType] + 1
+	}
+}
+
+func (s eventStats) publishStats() {
+	for idx, count := range s.eventCount {
+		eventType := eventspb.BusEventType(idx)
+		if count > 0 {
+			switch eventType {
+			case eventspb.BusEventType_BUS_EVENT_TYPE_UNSPECIFIED:
+				metrics.PublishedEventsAdd("EventBus-Unspecified", float64(count))
+			case eventspb.BusEventType_BUS_EVENT_TYPE_ALL:
+				metrics.PublishedEventsAdd("EventBus-All", float64(count))
+			case eventspb.BusEventType_BUS_EVENT_TYPE_TIME_UPDATE:
+				metrics.PublishedEventsAdd("EventBus-TimeUpdate", float64(count))
+			case eventspb.BusEventType_BUS_EVENT_TYPE_TRANSFER_RESPONSES:
+				metrics.PublishedEventsAdd("EventBus-TransferResponses", float64(count))
```

Comment:
Maybe you could use the map ```
BusEventType_name = map[int32]string{
		0:   "BUS_EVENT_TYPE_UNSPECIFIED",
		1:   "BUS_EVENT_TYPE_ALL",
```
Instead of having to enumerate them all?

Reply:
thanks just noticed that

Reply:
sorted and made some changes to the subscriber count for the event but to log subscription by type